types/convert.go

```diff
@@ -137,16 +137,8 @@ func ConvertFloatToUint(sc *stmtctx.StatementContext, fval float64, upperBound u
 		}
 		return uint64(int64(val)), overflow(val, tp)
 	}
-	ubf := float64(upperBound)
-	// Because math.MaxUint64 can not be represented precisely in iee754(64bit),
-	// so `float64(math.MaxUint64)` will make a num bigger than math.MaxUint64,
-	// which can not be represented by 64bit integer.
-	// So `uint64(float64(math.MaxUint64))` is undefined behavior.
-	if val == ubf {
-		return uint64(math.MaxInt64), nil
-	}
-	if val > ubf {
-		return uint64(math.MaxInt64), overflow(val, tp)
+	if val > float64(upperBound) {
+		return upperBound, overflow(val, tp)
```

Comment:
Any reason for this change?

Reply:
This is not a origin change in #12858，it is a conflict code in auto cherry-pick so i remove it， click `changes for all commits` can see this。